### ISSUE TITLE
Trying to open Treemacs for projectile root produces `wrong type argument: arrayp, nil` message

### ISSUE THREAD

**Issue by dtluna** [syl20bnr/spacemacs#12601]
#### Description :octocat:
Treemacs for projectile root doesn't work.

#### Reproduction guide :beetle:
- Start Emacs
- Open a file in a project
- Try to open Treemacs and add current projectile root to its list of projects (`SPC p t`)

*Observed behaviour:* :eyes: :broken_heart:
No Treemacs window is opened and you get a `wrong type argument: arrayp, nil` message at the bottom

*Expected behaviour:* :heart: :smile:
Treemacs window for the currect projectile root is opened on the left

#### System Info :computer:
- OS: gnu/linux
- Emacs: 26.2
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 50203d2c0)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: ivy
- Layers:
```elisp
(yaml python javascript elixir html rust ivy auto-completion emacs-lisp markdown
      (org :variables org-enable-reveal-js-support t org-enable-github-support t)
      (go :variables go-use-golangci-lint t)
      syntax-checking)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS GLIB NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD LCMS2

**Comment by duianto** [comment]
It's working as expected for me in both Ubuntu 18.04 and Windows 1903.
On the latest develop branch with updated packages.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 27.0.50
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 3df5b43d2)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(emacs-lisp git helm html markdown multiple-cursors org spell-checking syntax-checking treemacs version-control)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND GPM DBUS GSETTINGS GLIB NOTIFY INOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE HARFBUZZ M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS XWIDGETS LIBSYSTEMD PDUMPER LCMS2 GMP


#### System Info :computer:
- OS: windows-nt
- Emacs: 26.2
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 3df5b43d2)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(autohotkey dap
            (dart :variables dart-sdk-path "C:/tools/dart-sdk/" dart-enable-analysis t)
            emacs-lisp git helm
            (html :variables css-enable-lsp t)
            (javascript :variables)
            lsp markdown multiple-cursors org python ruby shell spell-checking syntax-checking treemacs version-control)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS THREADS LCMS2

**Comment by dtluna** [comment]
@duianto I've pulled the latest changes, updates the packages, added `treemacs` to the list of layers and the message has changed to `wrong type argument: arrayp, (invalid-path "Path is nil")`
I'm running Arch Linux FYI.

**Comment by dtluna** [comment]
@duianto I'll install the `emacs-git` package from AUR which provides Emacs version `27.0.50.137996-1` and see if everything works there

**Comment by dtluna** [comment]
@duianto Yes, it works just fine in Emacs `27.0.50`

**Comment by Trapez** [comment]
I get the same error as @dtluna on current develop branch.

**Comment by duianto** [comment]
>Open a file in a project

That's what I did before pressing `SPC p t`.

The `path is nil` message is shown when `SPC p t` is pressed in a buffer that isn't part of a project.

We'll discus a solution in Trapez's PR:
Fix treemacs-project-toggle (SPC p t) #12614
